modules/c/nitf/source/ImageIO.c

```diff
@@ -7165,6 +7169,7 @@ NITFPRIV(int) nitf_ImageIO_writeToBlock(_nitf_ImageIOBlock * blockIO,
                              NITF_STRERROR(NITF_ERRNO));
             return NITF_FAILURE;
         }
+        memset(blockCntl->block, 0, nitf->blockSize);
```

Comment:
This is just a big scratch buffer right?  So I don't think we need to actually `memset` it upfront...

Reply:
That's what I assumed, but valgrind complains a lot without it.

    ==29705== Conditional jump or move depends on uninitialised value(s)
    ==29705==    at 0xCBABE71: opj_t1_encode_cblk (t1.c:2288)
    ==29705==    by 0xCBABC79: opj_t1_encode_cblks (t1.c:2192)
    ==29705==    by 0xCB8338D: opj_tcd_t1_encode (tcd.c:2505)
    ==29705==    by 0xCB80DDE: opj_tcd_encode_tile (tcd.c:1408)
    ==29705==    by 0xCB5E4E9: opj_j2k_write_sod (j2k.c:4680)
    ==29705==    by 0xCB6F459: opj_j2k_write_first_tile_part (j2k.c:11754)
    ==29705==    by 0xCB6ECCD: opj_j2k_post_write_tile (j2k.c:11513)
    ==29705==    by 0xCB6FEA4: opj_j2k_write_tile (j2k.c:12108)
    ==29705==    by 0xCB574EE: opj_write_tile (openjpeg.c:916)
    ==29705==    by 0xCB3EA5D: OpenJPEGWriter_setTile (OpenJPEGImpl.c:1028)
    ==29705==    by 0xCB3F91B: j2k_Writer_setTile (Writer.c:57)
    ==29705==    by 0xCB3C49D: implWriteBlock (J2KCompress.c:376)

Reply:
Hmm... that indicates to me that we're not actually filling in some image content (or something else) fully before calling the compression.  Can you try to track down if the test program isn't fully filling in the input image or where this is coming from (may be easiest to set the buffer to 255 everywhere then try to find where/when/why you still see 255s by the time you get to compression)?

Reply:
Punting on this one for now - will address in future PR